sycl/source/detail/graph_impl.cpp

```diff
@@ -374,16 +374,18 @@ sycl::event exec_graph_impl::enqueue(
   if (CommandBuffer) {
     NewEvent = CreateNewEvent();
     RT::PiEvent *OutEvent = &NewEvent->getHandleRef();
-
-    // If we have no requirements for accessors for the command buffer, enqueue
-    // it directly
-    if (MRequirements.empty()) {
+    // Merge requirements from the nodes into requirements (if any) from the
+    // handler.
+    CGData.MRequirements.insert(CGData.MRequirements.end(),
+                                MRequirements.begin(), MRequirements.end());
+    // If we have no requirements or dependent events for the command buffer,
+    // enqueue it directly
+    if (CGData.MRequirements.empty() + CGData.MEvents.size()) {
```

Comment:
I'm not sure I understand the condition here as it appears to me that the condition is not in line with the comment.
Do we need an “or” condition here or rather a logical “and”?
Last point: maybe the design doc should be updated accordingly because there is no mention of an conditions on dependent events for the enqueuing process:  https://github.com/reble/llvm/blob/e451e65a02e3c4876d6564f611c62cdb6a8ef83e/sycl/doc/design/CommandGraph.md#scheduler-integration

Reply:
You're correct this was an error and is fixed now, thanks!